src/Opaleye/Internal/HaskellDB/PrimQuery.hs

```diff
@@ -60,6 +60,7 @@ data BinOp      = (:==) | (:<) | (:<=) | (:>) | (:>=) | (:<>)
                 | (:->) | (:->>) | (:#>) | (:#>>)
                 | (:@>) | (:<@) | (:?) | (:?|) | (:?&)
                 | (:&&) | (:<<) | (:>>) | (:&<) | (:&>) | (:-|-)
+                | OpArrayIndex
```

Comment:
It seemed natural to me following precedent set by `UnOp`, note that `UnOp` is considered one `PrimExpr` although it comes in various syntactical forms (function, prefix, postfix). Each of these forms are then broken down into corresponding `SqlExpr`s in the `defaultSqlExpr` function.

Why should binary operations, in their different syntactical forms (infix, array subscript), follow a different pattern?